mlir/lib/Gradient/Transforms/BufferizationPatterns.cpp

```diff
@@ -107,10 +130,16 @@ class BufferizeBackpropOp : public OpConversionPattern<BackpropOp> {
         generateAllocations(rewriter, loc, calleeResults, resShadows);
 
         DenseIntElementsAttr diffArgIndicesAttr = adaptor.getDiffArgIndices().value_or(nullptr);
-        rewriter.create<BackpropOp>(loc, TypeRange{}, op.getCalleeAttr(), adaptor.getArgs(),
-                                    argShadows, calleeResults, resShadows, diffArgIndicesAttr);
+        auto bufferizedBackpropOp = rewriter.create<BackpropOp>(
+            loc, scalarReturnTypes, op.getCalleeAttr(), adaptor.getArgs(), argShadows,
```

Comment:
Just to make sure why we init the backprop op with results here. Is it because we only do the bufferization for the tensors and we still need the float as results?

Reply:
Yes, exactly.